### PR TITLE
 Build test packages for windows, linux-x64, linux-arm, linux-arm64 and macOS

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Create test packages for windows, linux and macOS. 

## PR Context

We need the test packages for release automation tests for all the platforms that we support.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
@PoshChan Please get test failures

**Comment by adityapatwardhan** [comment]
@PoshChan please try windows

**Comment by PoshChan** [comment]
@adityapatwardhan, I do not understand: please try windows

Commands available in this repo for you:
  - `retry <target>` this will attempt to retry only the failed jobs for the target pipeline
  - `rebuild <target>` this will perform a complete rebuild of the target pipeline, `rerun` can be used in place of `rebuild`
    Supported values for \<target\> which can be a comma separated list are: `static`,`windows`,`macos`,`linux`,`all`
  - `get failures` this will attempt to get the latest failures for all of the target pipelines
  - `remind me in <value> <units>` this will create a reminder that will be posted after the specified duration
    \<value\> is a number, and \<units\> can be `minutes`, `hours`, or `days` (singular or plural)

File: build.psm1 [hunk 1]
```diff
@@ -842,7 +842,9 @@ function Get-PesterTag {
 
 function Publish-PSTestTools {
     [CmdletBinding()]
-    param()
+    param(
+        $runtime
```
Comment (TravisEz13) [inline build.psm1]:
```suggestion
        [string]
        $runtime
```

File: tools/releaseBuild/azureDevOps/templates/testartifacts.yml [hunk 2]
```diff
@@ -16,26 +12,35 @@ jobs:
 
   - pwsh: |
       Import-Module ./build.psm1
-      New-TestPackage -Destination $(System.ArtifactsDirectory)
-    displayName: Build test package
 
-  - pwsh: |
-      if (-not (Test-Path $(System.ArtifactsDirectory)/TestPackage.zip))
+      function BuildTestPackage($runtime)
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/templates/testartifacts.yml]:
```suggestion
      function BuildTestPackage([string] $runtime)
```

File: tools/releaseBuild/azureDevOps/diagram.puml [hunk 3]
```diff
@@ -39,6 +39,8 @@ folder "Package Signing and Upload" as PkgSignUpload {
 folder "Build Test Artifacts" as TestArtifacts {
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/diagram.puml]: Can you add `TestArtifacts` to start and finish?

Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/diagram.puml]: It already is connected to start and finish.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**